include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp

```diff
@@ -0,0 +1,173 @@
+    {
+        reset_optimum();
+    }
+    //!\}
+
+    /*!\brief Tracks any cell within the alignment matrix.
+     *
+     * \tparam cell_t The cell type of the alignment matrix; must have a member function `optimal_score()`.
+     *
+     * \param[in] cell The current cell to be tracked.
+     *
+     * \returns The forwarded cell.
+     *
+     * \details
+     *
+     * A call to this function only tracks the optimal score of the given cell if the configuration of the alignment
+     * algorithm requires it, for example when a local alignment shall be computed.
+     */
+    template <typename cell_t>
+    decltype(auto) track_cell(cell_t && cell) noexcept
+    {
+        return std::forward<cell_t>(cell);
+    }
+
+    /*!\brief Tracks the last cell of a row within the alignment matrix.
+     *
+     * \tparam cell_t The cell type of the alignment matrix; must have a member function `optimal_score()`.
+     *
+     * \param[in] cell The current cell to be tracked.
+     *
+     * \returns The forwarded cell.
+     *
+     * \details
+     *
+     * A call to this function only tracks the optimal score of the given cell if the configuration of the alignment
+     * algorithm requires it, for example when a semi-global alignment shall be computed.
+     */
+    template <typename cell_t>
+    decltype(auto) track_last_row_cell(cell_t && cell) noexcept
+    {
+        return std::forward<cell_t>(cell);
+    }
+
+    /*!\brief Tracks the last cell of a column within the alignment matrix.
+     *
+     * \tparam cell_t The cell type of the alignment matrix; must have a member function `optimal_score()`.
+     *
+     * \param[in] cell The current cell to be tracked.
+     *
+     * \returns The forwarded cell.
+     *
+     * \details
+     *
+     * A call to this function only tracks the optimal score of the given cell if the configuration of the alignment
+     * algorithm requires it, for example when a semi-global alignment shall be computed.
+     */
+    template <typename cell_t>
+    decltype(auto) track_last_column_cell(cell_t && cell) noexcept
+    {
+        return std::forward<cell_t>(cell);
+    }
+
+    /*!\brief Tracks the final cell of the alignment matrix.
+     *
+     * \tparam cell_t The cell type of the alignment matrix; must have a member function `optimal_score()`.
+     *
+     * \param[in] cell The current cell to be tracked.
+     *
+     * \returns The forwarded cell.
+     *
+     * \details
+     *
+     * A call to this function only tracks the optimal score of the given cell if the configuration of the alignment
+     * algorithm requires it, for example when a global alignment shall be computed.
+     */
+    template <typename cell_t>
+    decltype(auto) track_final_cell(cell_t && cell) noexcept
+    {
+        m_optimal_score = (cell.optimal_score() >= m_optimal_score) ? cell.optimal_score() : m_optimal_score;
+        return std::forward<cell_t>(cell);
+    }
```

Comment:
Some things I don't understand yet:

- The other `track_...` function don't do anything yet because the new alignment code only works for global alignments so far, but you added the functions here already, correct?
- The documentation for `track_final_cell` indicates that the optimal score is only tracked if configured, but I don't see a dependency on the configuration here? 
- Can you give an example on how the other functions will differ from the `track_final_cell` in the future?

Reply:
>* The other track_... function don't do anything yet because the new alignment code only works for global alignments so far, but you added the functions here already, correct?

correct, to see that the interface of the algorithm already works with this and the vectorised global alignment needs some of them already.

>The documentation for track_final_cell indicates that the optimal score is only tracked if configured, but I don't see a dependency on the configuration here?

Yep, because later we evaluate the configuration object and check what parameters have been set for free-end gaps etc and then conditionally enable/disable the compare and exchange part in the respective functions.

> Can you give an example on how the other functions will differ from the track_final_cell in the future?

For example:

```cpp
auto track_last_row_cell(cell)
{
   if (m_track_last_row_enabled)
      compare_and_exchange(cell);

   return ...;
}
```
Where compare_and_exchange will be the function implementing the actual comparison and update of the global optimum, which will be resused by all functions, or overwritten by specialisations of this policy.